components/base/wheeled/kinematics.go

```diff
@@ -51,24 +68,82 @@ func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
-	// TODO(RSDK-2311): complete the implementation
-	return []referenceframe.Input{}, errors.New("not implemented yet")
+	// TODO(rb): make a transformation from the component reference to the base frame
+	pose, _, err := kwb.slam.GetPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pt := pose.Point()
+	theta := math.Mod(pose.Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	return []referenceframe.Input{{Value: pt.X}, {Value: pt.Y}, {Value: theta}}, nil
 }
 
-func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// TODO(RSDK-2311): complete the implementation
-	return errors.New("not implemented yet")
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, desired []referenceframe.Input) (err error) {
+	// this loop polls the error state and issues a corresponding command to move the base to the objective
+	// when the base is within the positional threshold of the goal, exit the loop
+	// TODO(rb): check for the context being cancelled and stop if so
+	for {
+		current, err := kwb.CurrentInputs(ctx)
+		if err != nil {
+			return err
+		}
+
+		// get to the x, y location first
+		desiredHeading := math.Atan2(current[1].Value-desired[1].Value, current[0].Value-desired[0].Value)
+		if commanded, err := kwb.issueCommand(ctx, current, []referenceframe.Input{desired[0], desired[1], {desiredHeading}}); err != nil {
+			if commanded {
+				continue
+			}
+		}
+
+		// no command to move to the x, y location was issued, correct the heading and then exit
+		if commanded, err := kwb.issueCommand(ctx, current, []referenceframe.Input{current[0], current[1], desired[2]}); err != nil {
+			if !commanded {
+				return nil
+			}
+		}
+	}
 }
 
-// MakeModelFrame builds the kinematic model associated with the kinematicWheeledBase
-// Note that this model is not intended to be registered in the frame system.
-func MakeModelFrame(name string, collisionGeometry spatialmath.Geometry, limits []referenceframe.Limit) (referenceframe.Model, error) {
-	// build the model - SLAM convention is that the XY plane is the ground plane
-	frame2D, err := referenceframe.NewMobile2DFrame(collisionGeometry.Label(), limits, collisionGeometry)
+// issueCommand issues a relevant command to move the base to the given desired inputs and returns the boolean describing
+// if it issued a command successfully.  If it is already at the location it will not need to issue another command and can therefore
+// return a false.
+func (kwb *kinematicWheeledBase) issueCommand(ctx context.Context, current, desired []referenceframe.Input) (bool, error) {
+	distErr, headingErr, err := kwb.errorState(current, desired)
 	if err != nil {
-		return nil, err
+		return false, err
+	}
+	if distErr > distThresholdMM && math.Abs(headingErr) > headingThresholdDegrees {
+		return true, kwb.Spin(ctx, -headingErr, 60, nil) // base is headed off course; spin to correct
```

Comment:
What are these magic numbers?

Reply:
Just pushed something up to address this!